pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			assert!(
+				T::MaxInvulenrables::get() > (self.invulnerables.len() as u32),
+				"genesis invulnerables are more than T::MaxInvulenrables",
+			);
+			<Invulnerables<T>>::put(&self.invulnerables);
+		}
+	}
+
+	#[pallet::event]
+	#[pallet::metadata(T::AccountId = "AccountId", BalanceOf<T> = "Balance")]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event documentation should end with an array that provides descriptive names for event
+		/// parameters. [something, who]
+		NewInvulnerables(Vec<T::AccountId>),
+		NewAllowedAuthorCount(u32),
+		NewAuthorBond(BalanceOf<T>),
+		AuthorAdded(T::AccountId, BalanceOf<T>),
+		AuthorRemoved(T::AccountId)
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		TooManyAuthors,
+		Unknown,
+		Permission,
+		AlreadyAuthor,
+		NotAuthor
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	// These functions materialize as "extrinsics", which are often compared to transactions.
+	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000)]
+		pub fn set_invulnerables(
+			origin: OriginFor<T>,
+			new: Vec<T::AccountId>,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			if (new.len() as u32) > T::MaxInvulenrables::get() {
+				log::warn!(
+					"invulnerables > T::MaxInvulenrables; you might need to run benchmarks again"
+				);
+			}
+			<Invulnerables<T>>::put(&new);
+			Self::deposit_event(Event::NewInvulnerables(new));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn set_allowed_author_count(origin: OriginFor<T>, allowed: u32) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			if allowed > T::MaxAuthors::get() {
+				log::warn!(
+					"allowed > T::MaxAuthors; you might need to run benchmarks again"
+				);
+			}
+			<AllowedAuthors<T>>::put(&allowed);
+			Self::deposit_event(Event::NewAllowedAuthorCount(allowed));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn set_author_bond(origin: OriginFor<T>, author_bond: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			<AuthorBond<T>>::put(&author_bond);
+			Self::deposit_event(Event::NewAuthorBond(author_bond));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn register_as_author(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			// lock deposit to start or require min?
+			let who = ensure_signed(origin)?;
+			let deposit = <AuthorBond<T>>::get();
+			let length = <Authors<T>>::decode_len().unwrap_or_default();
+			ensure!((length as u32) < Self::allowed_authors(), Error::<T>::TooManyAuthors);
```

Comment:
```suggestion
			ensure!((length as u32) <= Self::allowed_authors(), Error::<T>::TooManyAuthors);
```

Reply:
so we are going to add one to the authors later, so if it is less then or equal and we have idk 5 people as authors with a max of 5, then this would allow 6, so we are checking that it is under 5 so the last person in can turn it from 4 -> 5

Reply:
Ah, :facepalm: